packages/mui-base/src/Slider/Root/useSliderRoot.ts

```diff
@@ -0,0 +1,485 @@
+ */
+function useSliderRoot(parameters: UseSliderParameters): UseSliderReturnValue {
+  const {
+    'aria-labelledby': ariaLabelledby,
+    defaultValue,
+    direction = 'ltr',
+    disabled = false,
+    largeStep = 10,
+    marks: marksProp,
+    max = 100,
+    min = 0,
+    minStepsBetweenValues = 0,
+    name,
+    onValueChange,
+    onValueCommitted,
+    orientation = 'horizontal',
+    rootRef,
+    scale = Identity,
+    step = 1,
+    tabIndex,
+    value: valueProp,
+  } = parameters;
+
+  // We can't use the :active browser pseudo-classes.
+  // - The active state isn't triggered when clicking on the rail.
+  // - The active state isn't transferred when inversing a range slider.
+  const [active, setActive] = React.useState(-1);
+  const [open, setOpen] = React.useState(-1);
+  const [dragging, setDragging] = React.useState(false);
+
+  const controlRef: React.MutableRefObject<HTMLElement | null> = React.useRef(null);
+
+  const registerSliderControl = React.useCallback((element: HTMLElement | null) => {
+    if (element) {
+      controlRef.current = element;
+    }
+  }, []);
+
+  const [valueState, setValueState] = useControlled({
+    controlled: valueProp,
+    default: defaultValue ?? min,
+    name: 'Slider',
+  });
+
+  const { contextValue: compoundComponentContextValue, subitems } = useCompoundParent<
+    string,
+    SliderThumbMetadata
+  >();
+
+  const handleValueChange = React.useCallback(
+    (value: number | number[], thumbIndex: number, event: Event | React.SyntheticEvent) => {
+      if (!onValueChange) {
+        return;
+      }
+
+      // Redefine target to allow name and value to be read.
+      // This allows seamless integration with the most popular form libraries.
+      // https://github.com/mui/material-ui/issues/13485#issuecomment-676048492
+      // Clone the event to not override `target` of the original event.
+      const nativeEvent = (event as React.SyntheticEvent).nativeEvent || event;
+      // @ts-ignore The nativeEvent is function, not object
+      const clonedEvent = new nativeEvent.constructor(nativeEvent.type, nativeEvent);
+
+      Object.defineProperty(clonedEvent, 'target', {
+        writable: true,
+        value: { value, name },
+      });
+
+      onValueChange(value, thumbIndex, clonedEvent);
+    },
+    [name, onValueChange],
+  );
+
+  const range = Array.isArray(valueState);
+
+  const values = React.useMemo(() => {
+    return (range ? valueState.slice().sort(asc) : [valueState]).map((val) =>
+      val == null ? min : clamp(val, min, max),
+    );
+  }, [max, min, range, valueState]);
+
+  const marks = React.useMemo(() => {
+    if (marksProp === true && step !== null) {
+      return [...Array(Math.floor((max - min) / step) + 1)].map((_, index) => ({
+        value: min + step * index,
+      }));
+    }
+
+    return marksProp ?? [];
+  }, [marksProp, max, min, step]);
+
+  const marksValues = (marks as Mark[]).map((mark: Mark) => mark.value);
```

Comment:
`const marks: Mark[] = React.useMemo(() => {`?

Reply:
I've removed `marks` completely for now, internally it was mostly being used for restricted values which will be redesigned in https://github.com/mui/base-ui/issues/412